R/calculate_study_day.R

```diff
@@ -0,0 +1,104 @@
+#' `calculate_study_day` performs study day calculation
+#' @description
+#' This function takes the an input data.frame and a reference data.frame (which
+#' is DM domain in most cases), and calculate the study day from reference date
+#' and target date. In case of unexpected conditions like reference date is not
+#' unique for each patient, or reference and input dates are not actual dates,
+#' NA will be returned for those records.
+#'
+#' @md
+#' @param ds_in input data.frame that contains the target date.
+#' @param ds_dm reference date.frame that contains the reference date.
+#' @param refdt reference date from `ds_dm` that will be used as reference to
+#' calculate the study day.
+#' @param tgdt target date from `ds_in` that will be used to calcualte the study
+#' day.
+#' @param study_day_var the new study day variable name in the output. For
+#' example, AESTDY for AE domain for CMSTDY for CM domain.
+#' @param merge_key character to represents the merging key between `ds_in` and
+#' `ds_dm`.
+#'
+#' @return a data.frame that takes all columns from `ds_in` and a new variable
+#' to represent the calculated study day.
+#'
+#' @examples
+#' \dontrun{
+#'   ae <- data.frame(
+#'     USUBJID = c("study123-123", "study123-124", "study123-125"),
+#'     AESTDTC = c("2012-01-01", "2012-04-14", "2012-04-14")
+#'   )
+#'   dm <- data.frame(
+#'     USUBJID = c("study123-123", "study123-124", "study123-125"),
+#'     RFSTDTC = c("2012-02-01", "2012-04-14", NA)
+#'   )
+#'   ae$AESTDTC <- as.Date(ae$AESTDTC)
+#'   dm$RFSTDTC <- as.Date(dm$RFSTDTC)
+#'   calculate_study_day(ae, dm, "RFSTDTC", "AESTDTC", "AESTDY")
+#' }
+#'
+#'
+
+calculate_study_day <- function(ds_in,
```

Comment:
REname this to sdtm_in

Reply:
done at https://github.com/pharmaverse/sdtm.oak/pull/20/commits/1333ca919099234514b26efe60fbaca0a370bb21